bin/prepare-release.sh

```diff
@@ -0,0 +1,151 @@
+SED="sed"
+
+WORKSPACE_ROOT="$( cd "$(dirname "$0")/.." ; pwd -P )"
+
+if [ -z "$1" ]; then
+    echo "Usage: $(basename "$0") <new version>"
+    echo
+    echo "Prepare for a new release by setting the version number"
+    exit 1
+fi
+
+NEW_VERSION="$1"
+DATE=$(date +%Y-%m-%d)
+
+if ! echo "$NEW_VERSION" | grep --quiet --extended-regexp '^[0-9]+\.[0-9]+\.[0-9]+(-[a-z0-9.-]+)?'; then
+    echo "error: Specified version '${NEW_VERSION}' doesn't match the Semantic Versioning pattern."
+    echo "error: Use MAJOR.MINOR.PATCH versioning."
+    echo "error: See https://semver.org/"
+    exit 1
+fi
+
+echo "Preparing update to v${NEW_VERSION} (${DATE})"
+echo "Workspace root: ${WORKSPACE_ROOT}"
+echo
+
+GIT_STATUS_OUTPUT=$(git status --untracked-files=no --porcelain)
+if [ -z "$ALLOW_DIRTY" ] && [ -n "${GIT_STATUS_OUTPUT}" ]; then
+    lines=$(echo "$GIT_STATUS_OUTPUT" | wc -l | tr -d '[:space:]')
+    echo "error: ${lines} files in the working directory contain changes that were not yet committed into git:"
+    echo
+    echo "${GIT_STATUS_OUTPUT}"
+    echo
+    echo 'To proceed despite this and include the uncommited changes, set the `ALLOW_DIRTY` environment variable.'
+    exit 1
+
+fi
+
+DOIT=y
+if [[ -n "$DRY_RUN" ]]; then
+    echo "Dry-run. Not modifying files."
+    DOIT=n
+fi
+
+### Rollup files ###
+
+# Update the version in the package.json file
+
+FILE=package.json
+run $SED -i.bak -E \
+    -e "s/^\"version\": \"[0-9a-z.-]+\"/\"version\": \"${NEW_VERSION}\"/" \
+    "${WORKSPACE_ROOT}/${FILE}"
+run rm "${WORKSPACE_ROOT}/${FILE}.bak"
+
+### Add-on manifest file ###
+
+# Update the version in the manifest.json file
+
+FILE=manifest.json
+run $SED -i.bak -E \
+    -e "s/^\"version\": \"[0-9a-z.-]+\"/\"version\": \"${NEW_VERSION}\"/" \
+    "${WORKSPACE_ROOT}/${FILE}"
+run rm "${WORKSPACE_ROOT}/${FILE}.bak"
+
+### Update package-log.json ###
+
+run npm i --package-lock-only
+
+### CHANGELOG ###
+
+FILE=CHANGELOG.md
+run $SED -i.bak -E \
+    -e "s/# Unreleased changes/# v${NEW_VERSION} (${DATE})/" \
+    -e "s/\.\.\.master/...v${NEW_VERSION}/" \
+    "${WORKSPACE_ROOT}/${FILE}"
+run rm "${WORKSPACE_ROOT}/${FILE}.bak"
+
+if [ "$DOIT" = y ]; then
+    CHANGELOG=$(cat "${WORKSPACE_ROOT}/${FILE}")
+    cat > "${WORKSPACE_ROOT}/${FILE}" <<EOL
+# Unreleased changes
+
+[Full changelog](https://github.com/mozilla-ion/ion-core-addon/compare/v${NEW_VERSION}...master)
+
+${CHANGELOG}
+EOL
+fi
+
+### Dependency summary ###
+
+echo "Updating dependency summary"
+run "${WORKSPACE_ROOT}"/bin/dependency-summary.sh
+
+echo "Everything prepared for v${NEW_VERSION}"
+echo
+echo "Changed files:"
+git status --untracked-files=no --porcelain || true
+echo
+echo "Create release commit v${NEW_VERSION} now? [y/N]"
+read -r RESP
+echo
+if [ "$RESP" != "y" ] && [ "$RESP" != "Y" ]; then
+    echo "No new commit. No new tag. Proceed manually."
+    exit 0
+fi
+
+run git add --update "${WORKSPACE_ROOT}"
+run git commit --message "Bumped version to ${NEW_VERSION}"
```

Comment:
Did you mean to `git tag` too?

Reply:
Nope: when cutting a new release through the GitHub UI a tag will be automatically created.